### ISSUE TITLE
Error logs

### ISSUE THREAD

**Issue by Weeeishy** [rogeraabbccdd/Kento-Rankme#24]
Hello,

These errors have been spamming my logs like crazy. It lags the server.

```
L 02/17/2017 - 19:04:37: [SM] Exception reported: Stack went below stack boundary
L 02/17/2017 - 19:04:37: [SM] Blaming: kento_rankme.smx
L 02/17/2017 - 19:04:37: [SM] Call stack trace:
L 02/17/2017 - 19:04:37: [SM]   [1] Line 1174, C:\Users\Yousef\Desktop\right\SM 1.8 28\addons\sourcemod\scripting\kento_rankme.sp::GetCurrentPlayers
L 02/17/2017 - 19:04:37: [SM]   [2] Line 1635, C:\Users\Yousef\Desktop\right\SM 1.8 28\addons\sourcemod\scripting\kento_rankme.sp::EventPlayerDeath
```
```
L 02/17/2017 - 19:06:13: [SM] Exception reported: Stack went below stack boundary
L 02/17/2017 - 19:06:13: [SM] Blaming: kento_rankme.smx
L 02/17/2017 - 19:06:13: [SM] Call stack trace:
L 02/17/2017 - 19:06:13: [SM]   [1] Line 1174, C:\Users\Yousef\Desktop\right\SM 1.8 28\addons\sourcemod\scripting\kento_rankme.sp::GetCurrentPlayers
L 02/17/2017 - 19:06:13: [SM]   [2] Line 1868, C:\Users\Yousef\Desktop\right\SM 1.8 28\addons\sourcemod\scripting\kento_rankme.sp::SalvarPlayer
L 02/17/2017 - 19:06:13: [SM]   [3] Line 2147, C:\Users\Yousef\Desktop\right\SM 1.8 28\addons\sourcemod\scripting\kento_rankme.sp::OnClientDisconnect
```

I had kento15 before updating today and it was working great with no errors.

I also noticed this error, but not a lot

```
L 02/17/2017 - 19:04:37: [SM] Exception reported: Invalid memory access
L 02/17/2017 - 19:04:37: [SM] Blaming: kento_rankme.smx
L 02/17/2017 - 19:04:37: [SM] Call stack trace:
L 02/17/2017 - 19:04:37: [SM]   [1] Line 496, C:\Users\Yousef\Desktop\right\SM 1.8 28\addons\sourcemod\scripting\include\string.inc::ExplodeString
L 02/17/2017 - 19:04:37: [SM]   [2] Line 995, C:\Users\Yousef\Desktop\right\SM 1.8 28\addons\sourcemod\scripting\kento_rankme.sp::OnSayText
```

**Comment by Weeeishy** [comment]
Also my servers started crashing randomly after updating the plugin to the latest version.

https://crash.limetech.org/6gy5mj5jzlns
https://crash.limetech.org/p5ijz5blcx3q
https://crash.limetech.org/y62eyohpa2tq
https://crash.limetech.org/qpzkdl7ylb6n
https://crash.limetech.org/uepxt2vqlfbp
https://crash.limetech.org/5nls3ypnrque
https://crash.limetech.org/5aa3kv7tavzt
https://crash.limetech.org/ueoxex4cb7ct

**Comment by Weeeishy** [comment]
UPDATE: Seems like this issue above was just temporary, its fixed itself apparently which is weird. I'll post some of the errors that I've encountered in my logs after cleaning them from the errors that I've posted above.

```
L 02/17/2017 - 21:44:34: [SM] Exception reported: Client 2 is not connected
L 02/17/2017 - 21:44:34: [SM] Blaming: kento_rankme.smx
L 02/17/2017 - 21:44:34: [SM] Call stack trace:
L 02/17/2017 - 21:44:34: [SM]   [0] IsFakeClient
L 02/17/2017 - 21:44:34: [SM]   [1] Line 1592, C:\Users\Yousef\Desktop\right\SM 1.8 28\addons\sourcemod\scripting\kento_rankme.sp::Event_BombExploded
```

```
L 02/17/2017 - 22:34:16: [SM] Exception reported: Invalid query Handle c3a1c590 (error: 4)
L 02/17/2017 - 22:34:16: [SM] Blaming: kento_rankme.smx
L 02/17/2017 - 22:34:16: [SM] Call stack trace:
L 02/17/2017 - 22:34:16: [SM]   [0] SQL_FetchString
L 02/17/2017 - 22:34:16: [SM]   [1] Line 1390, C:\Users\Yousef\Desktop\right\SM 1.8 28\addons\sourcemod\scripting\include\kento_rankme/cmds.inc::SQL_RankCallback
```

```
L 02/17/2017 - 19:44:32: [SM] Exception reported: Client 4 is not connected
L 02/17/2017 - 19:44:32: [SM] Blaming: kento_rankme.smx
L 02/17/2017 - 19:44:32: [SM] Call stack trace:
L 02/17/2017 - 19:44:32: [SM]   [0] GetClientName
L 02/17/2017 - 19:44:32: [SM]   [1] Line 2445, C:\Users\Yousef\Desktop\right\SM 1.8 28\addons\sourcemod\scripting\kento_rankme.sp::RankConnectCallback
L 02/17/2017 - 19:44:32: [SM]   [3] Call_Finish
L 02/17/2017 - 19:44:32: [SM]   [4] Line 869, C:\Users\Yousef\Desktop\right\SM 1.8 28\addons\sourcemod\scripting\kento_rankme.sp::CallRankCallback
L 02/17/2017 - 19:44:32: [SM]   [5] Line 842, C:\Users\Yousef\Desktop\right\SM 1.8 28\addons\sourcemod\scripting\kento_rankme.sp::SQL_GetRankCallback
```

Thank you!

**Comment by rogeraabbccdd** [comment]
The lastest sourcemod stable build is 1.8-5973, please try to update your sourcemod.
Your error is same as #15, and I will fix it soon.

**Comment by Weeeishy** [comment]
Server is still crashing with the latest stable sm version + latest version of your plugin.

https://crash.limetech.org/vkazjqmzrsts

Errors logs is filled with the 2 first errors on the first post. Might roll-back the rankme version to kento15

**Comment by rogeraabbccdd** [comment]
I tested today and I know why your server was crash.
Pull request #23 breaks my plugin.
I think I need to test it before I merge it to master.

**Comment by rogeraabbccdd** [comment]
I have update my plugin to 3.0.3.Kento.21 and I think I fix your problems.
Please test it if you have time.

**Comment by Weeeishy** [comment]
That's what I thought actually. After reverting back to 15 it seemed ok so I started to test every version after that, now Im at 20 and everything is ok. I will test out the version you just pushed. 

Thank you for your time man.

**Comment by Weeeishy** [comment]
Crashes and errors have been fixed with the latest version 3.0.3.Kento.22!

Thank you.
